test/appengine/test.js

```diff
@@ -93,97 +113,273 @@ if (process.env.TRAVIS_NODE_VERSION !== 'stable') {
-          if (code !== 0) {
-            done(new Error(sample.dir + ': failed to install dependencies!'));
-          } else {
-            done();
-          }
-        }
-      });
+  it('should deploy all samples', function (done) {
+    // 30 minutes because deployments are slow
+    this.timeout(30 * 60 * 1000);
+
+    testDeployments(done);
+  });
+});
+
+function testInstallation(sample, done) {
+
+  // Keep track off whether "done" has been called yet
+  var calledDone = false;
+
+  var proc = spawn('npm', ['install'], {
+    cwd: getPath(sample.dir)
+  });
+
+  proc.on('error', finish);
+
+  if (!process.env.TRAVIS) {
+    proc.stderr.on('data', function (data) {
+      console.log('stderr: ' + data);
     });
+  }
 
-    it(sample.dir + ': should return 200 and Hello World', function (done) {
-      var timeoutId;
-      var intervalId;
-      var success = false;
+  proc.on('exit', function (code) {
+    if (code !== 0) {
+      finish(new Error(sample.dir + ': failed to install dependencies!'));
+    } else {
+      finish();
+    }
+  });
+
+  // Exit helper so we don't call "cb" more than once
+  function finish(err) {
+    if (!calledDone) {
+      calledDone = true;
+      done(err);
+    }
+  }
+}
+
+function testLocalApp(sample, done) {
+  var calledDone = false;
+
+  var proc = spawn(sample.cmd, [sample.arg1], {
+    cwd: getPath(sample.dir)
+  });
+
+  proc.on('error', finish);
+
+  if (!process.env.TRAVIS) {
+    proc.stderr.on('data', function (data) {
+      console.log('stderr: ' + data);
+    });
+  }
+
+  proc.on('exit', function (code, signal) {
+    if (code !== 0 && signal !== 'SIGKILL') {
+      return finish(new Error(sample.dir + ': failed to run!'));
+    } else {
+      return finish();
+    }
+  });
+
+  // Give the server time to start up
+  setTimeout(function () {
+    // Test that the app is working
+    testRequest('http://localhost:8080', sample, function (err) {
+      proc.kill('SIGKILL');
+      return finish(err);
+    });
+  }, 5000);
+
+  // Exit helper so we don't call "cb" more than once
+  function finish(err) {
+    if (!calledDone) {
+      calledDone = true;
+      done(err);
+    }
+  }
+}
+
+function testDeployments(done) {
+
+  // Only deploy samples that have a projectId
+  var samplesToDeploy = sampleTests.filter(function (sample) {
+    return (sample.projectId &&
+      process.env.TRAVIS_NODE_VERSION === sample.TRAVIS_NODE_VERSION);
+  });
+
+  // Create deployment tasks
+  var tasks = samplesToDeploy.map(function (sample) {
+    return function (cb) {
+      // Keep track off whether "cb" has been called yet
       var calledDone = false;
 
-      var proc = spawn(sample.cmd, [sample.arg1], {
-        cwd: getPath(sample.dir)
+      var _cwd = getPath(sample.dir);
+      var args = [
+        'preview',
+        'app',
+        'deploy',
+        'app.yaml',
+        // Skip prompt
+        '-q',
+        '--project',
+        sample.projectId,
+        '--promote',
```

Comment:
promote will be removed soon and since you're deploying to a particular version you should just hit `https://version-dot-app-id.appspot.com`.